src/proc/pointcloud.cpp

```diff
@@ -115,13 +114,19 @@ namespace librealsense
 
     void pointcloud::inspect_other_frame(const rs2::frame& other)
     {
-        if (_other_stream != nullptr && other.get_profile().as<rs2::video_stream_profile>() == *_other_stream.get())
+        if (_stream_filter != _prev_stream_filter)
+            _other_stream = nullptr;
+        _prev_stream_filter = _stream_filter;
+
+        if (_extrinsics.has_value() && _other_stream != nullptr && other.get_profile().as<rs2::video_stream_profile>() == *_other_stream.get())
             return;
 
+        _other_stream = nullptr;
```

Comment:
The "if"  in the next line is redundant now, and should be removed to make it straight-forward: i.e:
_other_stream = <override_value>